### ISSUE TITLE
Can learn.fit_one_cycle(...) report training error rate?

### ISSUE THREAD

**Issue by kechan** [fastai/fastai#1691]
I reviewed the arguments, it doesn't look possible to have this report the training error rate? In Keras, this happens by default (they report training accuracy). 

Although it is almost always overfitting, seeing the training error rate goes to 0.000 is strangely satisfying. It is another indication your model has not HIGH bias. 

Is this available?

**Comment by sgugger** [comment]
No metrics are computed on the validation set only as this is bad practice to compute them on the training set. This is a conscious choice on our side. 
There is probably a workaround using a Callback that can add a new metrics (see `LossMetrics` if interested) but this isn't something we want in the library.
You can also use the function `validate` at the end of training, that takes a dataloader. Again though, metrics on the training set really don't mean anything.

**Comment by kechan** [comment]
I understood fastai is a pretty opinionated framework. But I can tell you that metrics on training can help you work more efficiently. I used that as an indication if your model has high bias. Or to stop the training when your training accuracy hit a 100%. 

Of course this may confuse beginner, and shouldn't be the default. I am surprised that it just wasn't considered to be an option.

**Comment by kechan** [comment]
Note: If I remembered right, Andrew Ng has a book where plots of train and val metrics are used to determined data complexity vs. accuracy, this is important in determining what size of dataset you may need. This is often overlooked if you use other's people dataset 'cos they come for free. If you have to collect your own, this is mighty pain in the ass and expensive. 

I have been using Keras last yr, and that training accuracy never mislead me, and helped me in this data requirement thing.

If you have a quick workaround with the callback, pls let me know. I haven't done any real work 'cos of feature mismatch with Keras. I have been trying to run the same thing on fastai and do a comparison, this is taking a bit longer than i planned.

**Comment by kechan** [comment]
FYI: This is a screenshot from his unpublished book. 

![img_0174](https://user-images.githubusercontent.com/122762/53192475-8ad80180-35dc-11e9-872b-8bb9fbe6fb6a.jpg)

So this plot can help you how your training set size vary with train/val metrics. Another thing, I sometimes found comparing the loss between train and val is apple and orange, and really hard to interprete. I thought I have seen ppl including vs excluding the weight decay loss. It is all just confusing. 

In all, the loss is sometimes considered as a proxy, to achieve good metrics. You only care about metrics at the end, not the loss, IMO. It's metrics that make you the $$.

**Comment by patricianing** [comment]
If use transforms in learner, will the metric reported computed on augmented validation dataset instead of the original validation dataset? If so, how to find out the total number of augmented dataset (for both training and validation) used in leaner?

**Comment by sgugger** [comment]
Please use the [forum](https://forums.fast.ai/) for questions related to the library. You'll probably find the answer to this one in an existing topic (but feel free to create one if you don't find it) and more generally, this is where the community is so you'll have a better chance at having an answer.
We only use GitHub for tracking standing bugs.
